contracts/test-utils/src/mocha_blockchain.ts

```diff
@@ -0,0 +1,187 @@
+import { BlockchainLifecycle } from '@0x/dev-utils';
+import { Web3ProviderEngine } from '@0x/subproviders';
+import { providerUtils } from '@0x/utils';
+import { TxData, Web3Wrapper } from '@0x/web3-wrapper';
+import * as _ from 'lodash';
+// Import ambient declarations (and clobber Jest).
+import 'mocha';
+import * as process from 'process';
+
+import { provider, txDefaults, web3Wrapper } from './web3_wrapper';
+
+// tslint:disable: no-namespace only-arrow-functions no-unbound-method
+
+export type ISuite = Mocha.ISuite;
+export type ISuiteCallbackContext = Mocha.ISuiteCallbackContext;
+export type SuiteCallback = (this: ISuiteCallbackContext) => void;
+export type ContextDefinitionCallback<T> = (description: string, callback: SuiteCallback) => T;
+export type BlockchainSuiteCallback = (this: ISuiteCallbackContext, env: BlockchainTestsEnvironment) => void;
+export type BlockchainContextDefinitionCallback<T> = (description: string, callback: BlockchainSuiteCallback) => T;
+export interface ContextDefinition extends Mocha.IContextDefinition {
+    optional: ContextDefinitionCallback<ISuite | void>;
+}
+
+/**
+ * Interface for `blockchainTests()`.
+ */
+export interface BlockchainContextDefinition {
```

Comment:
If the subfields of `resets` will always be the same as the outer fields, it possible to do something like this?
```
export interface BlockchainContextDefinitionPartial {
    only: BlockchainContextDefinitionCallback<ISuite>;
    skip: BlockchainContextDefinitionCallback<void>;
    optional: BlockchainContextDefinitionCallback<ISuite | void>;
    (description: string, callback: BlockchainSuiteCallback): ISuite;
}

export interface BlockchainContextDefinition extends BlockchainContextDefinitionPartial {
    resets: BlockchainContextDefinitionPartial;
}
```

Reply:
yeeerp :+1: